### PR TITLE
docs(frontend): Add introduction section to frontend docs

### PR DESCRIPTION
**Pull request by christian-konrad**
## Description

Closes https://github.com/camunda/camunda-docs/issues/3128

> [!IMPORTANT]  
> Stacked PR no. 1 for https://github.com/camunda/camunda-docs/issues/2616

▶️ Follow-up PR for merging into 8.4: https://github.com/camunda/camunda-docs/pull/3229

## When should this change go live?

- [x] This functionality is already available but undocumented.

## PR Checklist

- [x] Changes are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`).
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: docs/apis-tools/frontend-development/img/task-assignment.png
Comment (christinaausley) [inline docs/apis-tools/frontend-development/img/task-assignment.png:1]: Nitpick and non-blocking, but would love to see sentence-case throughout, i.e.  "Group task queue" and "Personal tasklist"

File: docs/apis-tools/frontend-development/introduction-to-task-applications.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+The **Processes** page lists all processes available to the logged in user, and allows the user to start a process from there.
+
+<img src={TasklistProcessesImg} style={{width: 800}} alt="Processes page in Tasklist" />
+
+Potential extensions are dependent on your use case. You can consider grouping processes by apps, domains, or teams, showing a process history, or adding a list of open process instances or cases.
+
+Learn more about the **Processes** page in the [Tasklist documentation](/components/tasklist/userguide/starting-processes.md).
+
+<!-- TODO move this to "concepts" -->
```
Comment (akeller) [inline docs/apis-tools/frontend-development/introduction-to-task-applications.md:89]: Is this ready for review? I see a TODO HTML comment.

File: docs/apis-tools/frontend-development/introduction-to-task-applications.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+## What are task applications?
+
+Task applications are end-user applications that allow humans to perform work orchestrated with a process. A [user task](/components/modeler/bpmn/user-tasks/user-tasks.md/#user-task-forms) (also referred to as a **human task**) represents a single **work item** to be performed by an individual or a group. The jobs of a task application include:
+
+- Listing available tasks and allowing users to select a task to work on.
+- Providing filter and search options for users so they can more easily find the right next task to work on.
+- Presenting the selected task and an interface for completing the task, usually via a form.
+- Providing an interface to create new tasks, e.g. by starting a new process.
+- Provide insight into the progress of work tasks, including processes and cases.
```
Comment (nathansandi) [inline docs/apis-tools/frontend-development/introduction-to-task-applications.md:27]: Should be 'insights' (plural) here?

Comment (nathansandi) [inline docs/apis-tools/frontend-development/introduction-to-task-applications.md:27]: [Suggestion] "Offering insights into the progress of work tasks, encompassing processes and cases.""

File: docs/apis-tools/frontend-development/introduction-to-task-applications.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+## Tasklist layout
+
+Camunda 8 comes with a ready-to-use Tasklist UI that implements all key concepts of a task application. The Tasklist UI is a generic task application; your custom task application should probably be tailored to your specific use case and also include external data sources and tools.
+
+The Tasklist UI is split into two main pages: the [tasks page](#task-page) and the [processes page](#processes-page).
+
+### Task page
+
+The task page lists all tasks pending for a user or user group, and allows users to pick and claim a task from that queue to work on. On the same page, the details of a selected task are displayed including the form that the user must submit in order to execute and complete the task.
```
Comment (nathansandi) [inline docs/apis-tools/frontend-development/introduction-to-task-applications.md:45]: Should we add some sentence talking about the possibility to unassign/assign a task?

**Comment by akeller** [comment]
@christian-konrad can we review and cleanup/merge as-is and follow up with well-scoped issues? I want to get this in 8.4 for you, but I wasn't sure if you want to address @nathansandi's feedback yourself.

**Comment by akeller** [comment]
Removing from 8.4 docs release scope.

File: docs/apis-tools/frontend-development/introduction-to-task-applications.md (hunk 1 above)
Comment (christian-konrad) [reply docs/apis-tools/frontend-development/introduction-to-task-applications.md:89]: The TODO is for a follow-up PR. Should we remove them from the HTML source? Are comments output in the built HTML?

File: docs/apis-tools/frontend-development/introduction-to-task-applications.md (hunk 3 above)
Comment (christian-konrad) [reply docs/apis-tools/frontend-development/introduction-to-task-applications.md:45]: I will do this in a follow-up PR

**Review by akeller — APPROVED** [review]

File: docs/apis-tools/frontend-development/introduction-to-task-applications.md (hunk 1 above)
Comment (akeller) [reply docs/apis-tools/frontend-development/introduction-to-task-applications.md:89]: They should not output, but I just want to ensure someone will follow up and not forget about them.

**Pull request merged**